### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by scottcreynolds**

### CODE REVIEW

**Comment by scottcreynolds** [comment]
@pletcher i think the linter needs an update? it accepts older versions of licence/contributing files with older copyright dates and spelling errors.

**Comment by scottcreynolds** [comment]
looking at the following lab, it seems you wanted this to go more in the way of the hybrid constructor/prototype pattern, so I'll do some rewriting.

File: README.md [hunk 1]
```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
...
+
+JavaScript uses prototypal OO. Instead of creating a nonfunctional class
+definition, we actually create the object, which is then used as a
+prototype to create other objects.
+
+In the real world, this would be like taking a bunch of laminated wood
+and assembling a Fthügënbøller, and then using that as your guide, or
+prototype, for creating other Fthügënbøllers, and Fthügënbøller-like objects.
+
+**//Flat-fact:** I'm regretting my choice to use impossible-to-type "Fthügënbøller" as my
```
Comment (pletcher) [inline README.md:30]: 😂

File: README.md [hunk 2]
```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
...
+made up example, but I'm committed to the bit. Besides, we all know
+everything about IKEA furniture is a nightmare.
+
+![crib](http://i.giphy.com/ZvkHeRqClNgUE.gif)
+
+So in both OO paradigms, we're defining the object, but in JavaScript,
```
Comment (pletcher) [inline README.md:36]: This is a great intro!

File: README.md [hunk 3]
```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
...
+
+To illustrate, try adding this code:
+
+```js
+Rectangle.prototype.internalAngles = 90;
+rect.internalAngles;
+square.internalAngles;
+```
+
+The `rect` can access `internalAngles`, but `square` cannot. Why? The
```
Comment (pletcher) [inline README.md]: This is really solid, but it's a little hard to follow. Might be important to really hold students' hands through this bit.

File: README.md [hunk 4]
```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
...
+
+// this works because the rect prototype is still Rectangle
+Rectangle.prototype.internalAngles = 90;
+rect.internalAngles;
+
+// but the square prototype is not Rectangle
+square.internalAngles;
+```
+
+Even though we call the `Rectangle` constructor function inside the
```
Comment (pletcher) [inline README.md:297]: This is great -- I really like how it walks students through what's happening.

File: README.md [hunk 5]
```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
...
+```js
+var rect = Object.create(rectangle);
+
+console.log(rect);
+// empty Object {}
+
+console.log(rect.sides);
+// 4
+
+console.log(rect.__proto__);
```
Comment (pletcher) [inline README.md]: Should we use `getPrototypeOf()` instead? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto

File: README.md [hunk 6]
```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
...
+properties of `quadrilateral`, so that even when we get down to
+`square`, we can still use the prototype system to delegate our
+perimeter formula up to `quadrilateral`.
+
+Since a square is just an equilateral rectangle, our `square` can use
+everything from the `rectangle` prototype, and just extend it to
+specifically set one side-length value instead of a `width` and
+`height`.
+
+This is the basic pattern of an inheritance chain. A parent object is
```
Comment (pletcher) [inline README.md]: This is great!

File: README.md [hunk 7]
```diff
@@ -9,21 +9,621 @@ JavaScript Inheritance Patterns
...
+more general than a child, which takes attributes from the parent and
+adds more specific attributes of its own.
 
 ## hasOwnProperty()
 
-- Make students "feel the pain" -- iterate through an object's properties and show that stuff is in there that shouldn't be (if it has inherited from another object)
+One interesting challenge with objects and inheritance is that we
```
Comment (pletcher) [inline README.md]: I was perhaps over-zealous here -- would it be easier to have a separate readme for hasOwnProperty()?

**Comment by pletcher** [comment]
Yeah, the linter is a bummer -- I think we can just ignore it for the moment.

**Comment by scottcreynolds** [comment]
@pletcher okay took another stab
